modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SelectVocabularies.js

```diff
@@ -149,18 +75,61 @@ function VocabularyTree({
 			showExpanderOnHover={false}
 		>
 			{(item) => (
-				<TreeView.Item key={item.externalReferenceCode}>
-					<div className="treeview-link-site-row">
-						<SiteRow
-							name={
-								item.descriptiveName_i18n?.[
-									Liferay.ThemeDisplay.getLanguageId()
-								] || item.descriptiveName
-							}
-							onSelect={_handleSelect}
-							vocabularies={item.children}
-						/>
-					</div>
+				<TreeView.Item
+					actions={
+						<div>
+							<ClayButton
+								aria-label={Liferay.Language.get('select-all')}
+								className="c-mr-1"
+								displayType="secondary"
+								onClick={(event) => {
+									event.preventDefault();
+
+									_handleSelect(item.children, true);
+								}}
+								onKeyDown={(event) => {
+									if (event.key === 'Enter') {
```

Comment:
Should the "space" key be used to select as well? Buttons usually work with both space and enter.

Reply:
Added some changes to support space as well 👍